### ISSUE TITLE
Google maps utils compiler error in new Xcode Version 15.0.1 

### ISSUE THREAD

**Issue by michalakakis** [googlemaps/google-maps-ios-utils#458]
I update the xcode to it's latest version and I came up with the following issue

Failed to build module 'GoogleMapsUtils'; this SDK is not supported by the compiler (the SDK is built with 'Apple Swift version 5.5.2 (swiftlang-1300.0.47.5 clang-1300.0.29.30)', while this compiler is 'Apple Swift version 5.9 (swiftlang-5.9.0.128.108 clang-1500.0.40.1)'). Please select a toolchain which matches the SDK.

**Comment by wangela** [comment]
If you would like to upvote the priority of this issue, please comment below or react on the original post above with :+1: so we can see what is popular when we triage.
--------

@michalakakis Thank you for opening this issue. 🙏
Please check out these other resources that might help you get to a resolution in the meantime:

- [Check the issue tracker](https://developers.google.com/maps/support/#issue_tracker) - bugs and feature requests for Google Maps Platform APIs and SDKs
- [Open a support case](https://developers.google.com/maps/support#contact-maps-support) - Get 1:1 support in Cloud Console.
- [Discord](https://discord.gg/fHTDmZe9En) - chat with other developers
- [StackOverflow](http://stackoverflow.com/questions/tagged/google-maps) - use the `google-maps` tag


_This is an automated message, feel free to ignore._

**Comment by markemer** [comment]
I'm seeing the same error with the XCFramework.

**Comment by piotrsmajek** [comment]
Same problem here, XCode 15.0.1 after migrating Google Maps SDK from XCFramework to SPM

**Comment by lluisgerard** [comment]
I had this issue recently and had to rollback my GoogleMaps to version 8.3.0. I commented it on this thread https://github.com/googlemaps/google-maps-ios-utils/issues/393#issuecomment-1843037835
Today I wanted to try if using SPM would allow me to upgrade to 8.3.1 but no, I have the same issue.

**Comment by lluisgerard** [comment]
To provide some details, I rolled back again my project to manual installation. The project that compiles successfully has:
- Google Maps 8.1.0
- Google Maps Utils 4.2.2

Is when I try to update to Google Maps version 8.3.0 (manually) or 8.3.1 (SPM) that I get the error `Failed to build module 'GoogleMapsUtils'; this SDK is not supported by the compiler`.

**Comment by federicopolesello** [comment]
Same issue to me, but using Xcode 14.3.0 (didn't try Xcode 15 yet).

Any update on this? I'm blocked due to this issue.
I'm using Xcode 14.3.0 and I installed GoogleMaps iOS SDK 8.3.1 through SPM.
Installing it through Cocoapods works but I'll would use the new SPM support.

![image](https://github.com/googlemaps/google-maps-ios-utils/assets/6029717/70ac3a2f-25aa-40f4-97f4-2cc8df13722a)

**Comment by andyzaharia** [comment]
Do we have any updates on this issue? The latest release of iOS GoogleMaps SDK is pointless because we cannot use it with the latest Google maps utils.

**Comment by Bilalyyy** [comment]
same problem, when will this be fixed??🫠😤

**Comment by chanffdavid** [comment]
Same problem here

**Comment by twprzybysz** [comment]
While https://github.com/googlemaps-samples/maps-sdk-for-ios-samples/issues/70 was solved, can we expect to resolve this issue around easter? 😉 @wangela

**Comment by RSady** [comment]
This update and Google Maps as whole is useless and trash.  My team is being force to go to native Apple Maps because of all the issues we've had with GMS.  Good riddance.

**Comment by WonKimGoRevel** [comment]
Same problem here.
- XCode 15.2
-  Google Maps 8.3.1
-  Google Maps Utils 4.2.2

Installing Google Maps through Cocoapods works.

**Comment by Mohsenkhodadadzadehthelist** [comment]
same issue here :
Google Maps Utils 4.2.2
installing through SPM

**Comment by MohammadJahirul** [comment]
<img width="681" alt="Screenshot 2024-02-06 at 11 52 30 AM" src="https://github.com/googlemaps/google-maps-ios-utils/assets/17243074/830cffbc-2287-496e-98ac-17d29c3097fc">

**Comment by lluisgerard** [comment]
Any updates on this? On December 16 someone from Google said that they were working on this.

**Comment by Jodebjo** [comment]
> While [googlemaps-samples/maps-sdk-for-ios-samples#70](https://github.com/googlemaps-samples/maps-sdk-for-ios-samples/issues/70) was solved, can we expect to resolve this issue around Easter? 😉 @wangela

I don't want to jinx it, but if I look at the rate google fixes, bugs we might still be wating in 2027.

I hope I am wrong...

**Comment by VladOrackle** [comment]
Temporary solution:
Point SPM to this branch https://github.com/googlemaps/google-maps-ios-utils/tree/wangela/8-0-0

**Comment by alexookah** [comment]
I tried the branch but it didn't work for me

![Screenshot 2024-03-06 at 22 38 24](https://github.com/googlemaps/google-maps-ios-utils/assets/9624843/22b24582-bb5b-486a-8f8e-14d13fb39377)

**Comment by dsmailes** [comment]
@alexookah is it possible you have the Google Maps SDK added to that project either by a custom SP or pod?

The branch being suggested has a built-in dependency on Google Maps and so you might be getting a duplicate copy of the framework.

**Comment by alexookah** [comment]
Yes the only way to have latest google maps sdk with google-maps utils is to have google maps with cocoapods and google-maps utils as spm

**Comment by dsmailes** [comment]
@alexookah you can, in my experience at least with this branch of Utils, remove the Google Maps pod and then use the official SPM version of Google Maps of your choice, or just allow the Utils package to manage via its dependency declared in package.swift

**Comment by VladOrackle** [comment]
My solution is for SPM usage only I guess
Utils having dependency on maps is quite warranted, because why would you want it without the maps

**Comment by alexookah** [comment]
There is another [issue](https://github.com/googlemaps/google-maps-ios-utils/issues/459) with SPM's
Currently with Xcode 15.3 we cant upload to TestFlight.

**Comment by ungurmaria** [comment]
any updates on this? For me the branch Vlad provided doesn't work.

**Comment by Kvadroman** [comment]
any updates or workarounds ?

**Comment by ungurmaria** [comment]
What appears to work for me now is having both Google Maps and google utils via SPM, BUT instead of using the branch mentioned by @VladOrackle, I used the last commit form this [branch](https://github.com/googlemaps/google-maps-ios-utils/tree/wangela/8-0-0), which as of now it's `de4e482df1fd729362db91239269e1e92cb0d787`

Below you can see my configuration. Maybe it helps someone. Working with the branch itself did not work for me and I still had the same error.

<img width="1082" alt="Screenshot 2024-03-11 at 12 33 38" src="https://github.com/googlemaps/google-maps-ios-utils/assets/23298298/401ba186-b434-4b00-84ac-8de3b1d06a18">

**Comment by Kvadroman** [comment]
> What appears to work for me now is having both Google Maps and google utils via SPM, BUT instead of using the branch mentioned by @VladOrackle, I used the last commit form this [branch](https://github.com/googlemaps/google-maps-ios-utils/tree/wangela/8-0-0), which as of now it's `de4e482df1fd729362db91239269e1e92cb0d787`
> 
> Below you can see my configuration. Maybe it helps someone. Working with the branch itself did not work for me and I still had the same error.
> 
> <img alt="Screenshot 2024-03-11 at 12 33 38" width="1082" src="https://private-user-images.githubusercontent.com/23298298/311671388-401ba186-b434-4b00-84ac-8de3b1d06a18.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAxNjExNTYsIm5iZiI6MTcxMDE2MDg1NiwicGF0aCI6Ii8yMzI5ODI5OC8zMTE2NzEzODgtNDAxYmExODYtYjQzNC00YjAwLTg0YWMtOGRlM2IxZDA2YTE4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzExVDEyNDA1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjN2ZmNTU1ZjNiZTE0YzgxODM1YTg5ZjY3MzVmZDczYTJmODViMjE5MTc3M2YxYzYxMWQ4MjhjMzdkZjBjMTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.v9o83fcmdrNk5hp9QTg9S003GBLq9vYkfYkHvUS2HJc">

but you need to update it manually every time
instead of this you may wrap in your own package and add this as dependency 

.package(name: "google-maps-ios-utils", url: "https://github.com/googlemaps/google-maps-ios-utils.git", branch: "wangela/8-0-0")

it updates every time when wangela/8-0-0 branch was updated

**Comment by Kvadroman** [comment]
`import PackageDescription

let package = Package(
    name: "YourPackageName",
    platforms: [.iOS(.v15)],
    products: [
        .library(
            name: "YourPackageName",
            targets: ["YourPackageName"]),
    ],
    dependencies: [
      .package(url: "https://github.com/googlemaps/ios-maps-sdk", .upToNextMinor(from: "8.4.0")),
      .package(url: "https://github.com/googlemaps/google-maps-ios-utils.git", .branch("wangela/8-0-0"))
    ],
    targets: [
        .target(
            name: "YourPackageName",
            dependencies: [
                .product(name: "GoogleMapsUtils", package: "google-maps-ios-utils"),
                .product(name: "GoogleMaps", package: "ios-maps-sdk"),
                .product(name: "GoogleMapsCore", package: "ios-maps-sdk"),
                .product(name: "GoogleMapsBase", package: "ios-maps-sdk")
            ],
            path: "Sources/YourPackageName"
        ),
        .testTarget(
            name: "YourPackageNameTests",
            dependencies: ["YourPackageName"]),
    ]
)`

I mean wrap this packages in one

![Снимок экрана 2024-03-11 в 23 24 41](https://github.com/googlemaps/google-maps-ios-utils/assets/86829286/6c7931a7-bdd6-41c3-9217-67ed0115a6c9)

all packages you need will downloaded automatically

**Comment by wangela** [comment]
Hi folks, still working on this in PR #444. Our goals are to update the Package to Apple's latest manifest structure, continue supporting CocoaPods, end support for Carthage, right-size version dependencies, and update the build, tests, and release actions based on source instead of binary packaging. It's almost done but there are some tests failing that I need to get resolved before I can be confident that the release will not break folks who install via either SPM or CocoaPods. Building from my branch is risky :)

**Comment by ungurmaria** [comment]
> Hi folks, still working on this in PR #444. Our goals are to update the Package to Apple's latest manifest structure, continue supporting CocoaPods, end support for Carthage, right-size version dependencies, and update the build, tests, and release actions based on source instead of binary packaging. It's almost done but there are some tests failing that I need to get resolved before I can be confident that the release will not break folks who install via either SPM or CocoaPods. Building from my branch is risky :)

Thanks for the response. Any estimate on when this will be merged? Would be really great to have support for SPM 🙇🏻

**Comment by salvatoreboemia** [comment]
Hi, any news on this topic?

when I install GoogleMaps and Places as SPM, GoogleMapsUtils (SPM) stop workings and I got the same error

**Comment by lluisgerard** [comment]
The branch `https://github.com/googlemaps/google-maps-ios-utils/tree/wangela/8-0-0` hasn't been updated in 2 months... how safe it is to use on production? 

I recently bumped my iOS version and I get another error now (Invalid Bundle), I need to fix this because of crashes on another dependency (I also suspect crashes on GoogleMaps that I don't get reports for).

**Comment by madswesti** [comment]
An update on the progress would be much appreciated @wangela

**Comment by piotrsmajek** [comment]
After 1 May it cans block release of our app... any update? @wangela

**Comment by salvatoreboemia** [comment]
Nothing works well in google maps / places / utils

**Comment by piotrsmajek** [comment]
I tried google maps and google maps utils as xcframework and I have the same issue.

XCode 15.2
Google Maps Utils version - 4.2.2
Google Maps - 8.5.0

![image](https://github.com/googlemaps/google-maps-ios-utils/assets/52935707/385fa3b4-3739-4e72-a1f5-c880de2a952c)

**Comment by rogerioth** [comment]
Same issue still remains;

- google-maps-ios-utils 4.2.2
- Google's ios-maps-sdk 8.4.0

Note that this is has the same cause: https://github.com/googlemaps/google-maps-ios-utils/issues/467


```Failed to build module 'GoogleMapsUtils'; this SDK is not supported by the compiler (the SDK is built with 'Apple Swift version 5.5.2 (swiftlang-1300.0.47.5 clang-1300.0.29.30)', while this compiler is 'Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)'). Please select a toolchain which matches the SDK.```


Portion of SPM Manifest:
```
    {
      "identity" : "ios-maps-sdk",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/googlemaps/ios-maps-sdk",
      "state" : {
        "revision" : "bd392d7d844b49ec6795871a3c25edd01ab839f2",
        "version" : "8.4.0"
      }
    },



    {
      "identity" : "google-maps-ios-utils",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/googlemaps/google-maps-ios-utils.git",
      "state" : {
        "revision" : "b56b39cec0e3d5938ddadbfeb4d3fec541ebbb5e",
        "version" : "4.2.2"
      }
    },
    {
      "identity" : "googleappmeasurement",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleAppMeasurement.git",
      "state" : {
        "revision" : "c7a5917ebe48d69f421aadf154ef3969c8b7f12d",
        "version" : "10.23.1"
      }
    },
    {
      "identity" : "googledatatransport",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleDataTransport.git",
      "state" : {
        "revision" : "a637d318ae7ae246b02d7305121275bc75ed5565",
        "version" : "9.4.0"
      }
    },
    {
      "identity" : "googleutilities",
      "kind" : "remoteSourceControl",
      "location" : "https://github.com/google/GoogleUtilities.git",
      "state" : {
        "revision" : "26c898aed8bed13b8a63057ee26500abbbcb8d55",
        "version" : "7.13.1"
      }
    },

```

**Comment by BassHunt** [comment]
Nada ainda? 
tem alguma resolução para esse problema?
alguma outra alternativa ? um quebra galho ? nada?

**Comment by piotrsmajek** [comment]
@wangela

**Comment by lluisgerard** [comment]
Could we please get an update on this issue? It's been ongoing for half a year now, and with Maps SDK for iOS version 9.0.0 already available, we were hoping for a corresponding Utils update.

Any information on the expected timeline would be greatly appreciated.

Thanks!

**Comment by rogerioth** [comment]
@googlemaps-bot any resource that can help with this persistent problem?

**Comment by lluisgerard** [comment]
@wangela Are there any updates? It seems like there might be a lack of resources for maintaining this repository. I understand there might be challenges behind the scenes, but it appears that either the project isn’t actively being worked on, or more developers are needed to manage it. If continuing support isn’t feasible, perhaps it would be better to deprecate the project rather than leaving it in this uncertain state. This limbo is becoming quite frustrating for the community. Thank you for your attention.

**Comment by salvatoreboemia** [comment]
Holaaaaaaa???

What do we need to do? 

remove completely Google and replace it with Apple Maps?

Please update this topic

Thanks

**Comment by carloshpdoc** [comment]
Hey @wangela. 

Wouldn't it be better to go back with the deleted branch wangela/8-0-0 until we have the fix 100% in the release?

**Comment by carloshpdoc** [comment]
I fixed this issue using the SDK version 5.0.0 and swift_version 5.3 after I used Swift 5.10.

**Comment by wangela** [comment]
v5.0.0 was released for SPM yesterday and CocoaPods today. v6.0.0 was released for both SPM and CocoaPods today.
See the [README](https://github.com/googlemaps/google-maps-ios-utils?tab=readme-ov-file#swift-package-manager) for installation instructions. Do not add dependencies on branches associated with pull requests.
